packages/utils/src/ast-utils/helpers.ts

```diff
@@ -8,14 +8,14 @@ type ObjectEntries<BaseType> = ObjectEntry<BaseType>[];
 export const isNodeOfType =
   <NodeType extends AST_NODE_TYPES>(nodeType: NodeType) =>
   (
-    node: TSESTree.Node | null | undefined,
+    node: null | TSESTree.Node | undefined,
   ): node is Extract<TSESTree.Node, { type: NodeType }> =>
     node?.type === nodeType;
 
 export const isNodeOfTypes =
   <NodeTypes extends readonly AST_NODE_TYPES[]>(nodeTypes: NodeTypes) =>
   (
-    node: TSESTree.Node | null | undefined,
+    node: null | TSESTree.Node | undefined,
```

Comment:
This looks worse to me, FWIW.

Reply:
I agree, which is why I still prefer (and use) `@typescript-eslint/sort-type-constituents` in my own projects even though it's deprecated. I'm hoping there's some way to achieve the same sorting with `eslint-plugin-perfectionist`, or that `@typescript-eslint/sort-type-constituents` gets brought back somehow... 😬

Reply:
Heh yeah. I also like having `null` and `undefined` at the end. IMO they're separate primitives that should be after the normal flow of defined types.

This is somewhat-coincidentally controllable by the [`ignoreCase` rule option](https://perfectionist.dev/rules/sort-union-types#ignorecase)... I'll apply that now.

It'd be nice if we could set the general settings for the plugin once - rather than having a setting for each rule... filed: https://github.com/azat-io/eslint-plugin-perfectionist/issues/202

Reply:
so `ignoreCase` fixes this particular case just by coincidence.

But it doesn't fix the issue generally, i.e.:
![image](https://github.com/user-attachments/assets/2d36a0a2-9090-45af-8df4-1968d9e1075f)

do you think that adding support for `null` and `undefined` at the end is something that `eslint-plugin-perfectionist` would add?

Reply:
Maybe! It'd be a good issue to file on them for sure.

Reply:
https://github.com/typescript-eslint/typescript-eslint/pull/9698#issuecomment-2296427550 🚀